xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp

```diff
@@ -2395,3 +2402,11 @@ void CAMLCodec::SetVideoRate(int videoRate)
   if (am_private)
     am_private->video_rate = videoRate;
 }
+
+int CAMLCodec::GetDecoderVideoRate()
+{
+  struct vdec_status vs;
+  m_dll->codec_get_vdec_state(&am_private->vcodec, &vs);
+  CLog::Log(LOGDEBUG, LOGVIDEO, "CAMLCodec::GetDecoderVideoRate: decoder status: w:%d h:%d f:%d e:%d s:%d", vs.width, vs.height, vs.fps, vs.error_count, vs.status);
+  return (int)(0.5 + ((float)UNIT_FREQ / (float)vs.fps));
```

Comment:
use C++ style casts

Reply:
Done

Reply:
The change I made yesterday to use C++ style casts has caused an issue where video can freeze for several seconds following seek and pause/play operations.

I'm just working on a solution to this - basically check whether the value being returned from this function is sensible - i.e. the rate returned is somewhere between 800 (120fps) and 19,200 (5fps).

Will submit a further PR to fix things up once I've had time to test it properly.